ddtrace/tracer/spancontext.go

```diff
@@ -229,10 +229,20 @@ func (t *trace) finishedOne() {
 	if len(t.spans) != t.finished {
 		return
 	}
+	t.finishTraceLocked(s)
+}
+
+func (t *trace) finishTraceLocked(last *span) {
 	if tr, ok := internal.GetGlobalTracer().(*tracer); ok {
 		// we have a tracer that can receive completed traces.
 		if t.priority != nil {
-			t.root.Metrics[keySamplingPriority] = *t.priority
+			// if the last span that finished is not the root, that means
+			// that the root has already finished as part of a separate,
+			// asynchronous operation, in which case we shouldn't touch
```

Comment:
Argh, this is also wrong. It's possible for a normal, sync trace to finish without having the root be the last.

e.g.

```
span1.start
span2.start
span1.end
span2.end
```
:|